src/PointwiseFunctions/AnalyticData/AnalyticData.hpp

```diff
@@ -0,0 +1,11 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+/// \cond
+namespace detail {
+// Empty base class for marking analytic data.
+struct MarkAsAnalyticData {};
```

Comment:
Calling this a `detail` doesn't seem appropriate.  It is used by every analytic data class and the type traits file.

Reply:
do you have an alternative namespace in mind, or just put it in the global namespace?

Reply:
I don't see any namespace that it would make sense in.  Global seems fine.

Reply:
These "mark as" classes are actually pretty much what I had in mind for "Protocols" in #1349. So how about we put these in `Protocols` namespaces that we handle like the `Tags` namespaces. Then they can go into `Protocols.hpp` files next to the `Tags.hpp` files. This also makes the `MarkAs` in the name redundant, so this could be e.g. `Protocols::AnalyticData`. Later we can add more functionality to protocols, e.g. make them better testable and add macros to generate the type traits.

Reply:
I don't think there's agreement on having "Protocols" in the code.

Reply:
That’s why I’m putting forward the suggestion :) the issue has been open for a long time.
What I’m saying is that these structs basically _are_ protocols, so we might as well name them as such and put them in appropriate files. I expect more will show up soon because it’s a useful metaprogramming tool and would be good to know where to put them.